### PR TITLE
Suppress doc tests when no test-related files are changed

### PR DESCRIPTION
**Pull request by frankbu**

### CODE REVIEW

File: prow/config/jobs/istio.io.yaml [hunk 1]
```diff
@@ -14,14 +14,20 @@ jobs:
     command: [entrypoint, prow/integ-suite-kind.sh, doc.test.profile_default]
     requirements: [kind]
     resources: 6Gi
+    trigger: run_if_changed
```
Comment (howardjohn) [inline prow/config/jobs/istio.io.yaml]: I think you just need the `regex` part

**Review by howardjohn — COMMENTED** [review]
`make gen` needed but lgtm

cc @ericvn

**Comment by ericvn** [comment]
I did run a regex checker against a set of possible changed files and it seems to work as desired, being positive for those changes we want to follow.

This is a good first step to prevent the running of tests when a change which shouldn't affect tests is merged, like a simple .md file.

**Review by ericvn — APPROVED** [review]

**Comment by ericvn** [comment]
A note of the files run against the checker. This doesn't have what was positive or not, but there where 18.
```
go.mod
go.sum
./archive/v1.13/zh/docs/setup/install/multicluster/multi-primary_multi-network/test.sh
./archive/v1.13/zh/docs/setup/install/multicluster/primary-remote/test.sh
./archive/v1.13/zh/docs/setup/install/multicluster/primary-remote_multi-network/
./content/en/docs/setup/additional-setup/getting-started/test.sh
./content/en/docs/setup/additional-setup/gateway/test.sh
./content/en/docs/setup/upgrade/canary/canary_upgrade_test.sh
./content/en/docs/setup/upgrade/canary/revision_tags_test.sh
./content/en/docs/setup/upgrade/helm/revlabels_test.sh
./content/en/docs/setup/upgrade/helm/inplace_test.sh
./content/en/docs/setup/upgrade/helm/canaryupgrade_test.sh
./content/en/boilerplates/snips/test-3.sh
./content/en/boilerplates/snips/test-1.sh
./scripts/test_status.sh
./tests
./tests/tests.mk
./tests/util/debug.sh
/archive/v1.13/zh/docs/examples/virtual-machines/snips.sh
./archive/v1.13/zh/docs/ops/configuration/mesh/config-resource-ready/snips.sh
./archive/v1.13/zh/docs/ops/configuration/mesh/app-health-check/snips.sh
./archive/v1.13/zh/docs/ops/configuration/telemetry/envoy-stats/snips.sh
./archive/v1.13/zh/docs/ops/configuration/traffic-management/network-topologies/snips.sh
./archive/v1.13/zh/docs/ops/diagnostic-tools/istioctl-analyze/snips.sh
./archive/v1.13/zh/boilerplates/snips
./archive/v1.13/boilerplates/snips
./content/en/docs/tasks/security/tls-configuration/workload-min-tls-version/snips.sh
./content/en/docs/tasks/security/cert-management/plugin-ca-cert/snips.sh
./content/en/docs/tasks/security/authorization/authz-tcp/snips.sh
./content/en/docs/tasks/security/authorization/authz-custom/snips.sh
./content/en/docs/tasks/security/authorization/authz-ingress/snips.sh
./content/en/docs/tasks/security/authorization/authz-dry-run/snips.sh
./content/en/docs/setup/upgrade/helm/index.md
```

**Pull request merged**